### PR TITLE
validation: Add `max_chain_depth` parameter to `ServerVerifier`

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
To get the remaining coverage.

### CODE REVIEW

File: src/rust/cryptography-x509-validation/src/lib.rs [hunk 1]
```diff
@@ -304,8 +304,7 @@ where
         // against the EE cert's SANs.
         self.policy.permits_leaf(leaf)?;
 
-        // NOTE: We start the chain depth at 1, indicating the EE.
-        let result = self.build_chain_inner(leaf, 1, true);
+        let result = self.build_chain_inner(leaf, 0, true);
```
Comment (tetsuo-cpp) [inline src/rust/cryptography-x509-validation/src/lib.rs:303]: We were previously subtracting one for the `pathlen` check. My thinking is that we should start from 0 and remove the subtraction. This way, the `pathlen` check and the `max_chain_depth` works the same way.

**Comment by tetsuo-cpp** [comment]
Ping @woodruffw. Not sure why "benchmark" is failing but this should give us the remaining coverage.

File: src/rust/cryptography-x509-validation/src/lib.rs (hunk 1 above)
Comment (woodruffw) [reply src/rust/cryptography-x509-validation/src/lib.rs:303]: Makes sense!

**Review by woodruffw — APPROVED** [review]

**Comment by woodruffw** [comment]
Yeah, I think the benchmark workflow is just broken on forks.

Nice work @tetsuo-cpp! Merging.

**Pull request merged**